vms/proposervm/service.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package proposervm
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/api"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/formatting"
+
+	avajson "github.com/ava-labs/avalanchego/utils/json"
+)
+
+type ProposerAPI struct {
```

Comment:
What is the reason for having the proposerVM as a field instead of just adding the functions to the proposerVM VM? 

If we wish to isolate the VM object then it makes sense to add two functions or (an) interface(s) with:

1. getLastAcceptedHeight
2. GetBlock 

and then it'll be easier to test.